src/pages/BDI_DataImport.page

```diff
@@ -83,7 +104,20 @@
                                 <div class="slds-form slds-form_stacked slds-m-left_large">
                                     <div class="slds-form-element">
                                         <div class="slds-form-element__control">
-                                            <apex:commandButton value="{!$Label.bdiRunBtn}" action="{!importData}" immediate="false" styleClass="slds-button slds-button_brand" rerender="idForm"/>
+                                            <apex:actionFunction name="importDataActionFunction"
+                                                                 action="{!importData}"
+                                                                 oncomplete="buttonsEnabled(true);"
+                                                                 rerender="idForm">
+                                            </apex:actionFunction>
+
+                                            <apex:commandButton value="{!$Label.bdiRunBtn}"
+                                                                action="{!importData}"
```

Comment:
Does having both the button action call {!importData} and the actionFunction call {!importData} mean that the controller method is being called twice? I haven't being in visualforce land in forever but it seems like that might be an issue

Reply:
Good point! I believe the action on the commandButton isn't actually doing anything at this point as returning false within its onclick function overrides the commandButton's normal behavior. So at the very least there's no point in having it there. Tested this out by dropping a sys debug in the controller's 'importData' method. I'm looking into your follow up comment below now. Currently under the impression that disabling the button before the post completes actually scrubs the request of which button was actually clicked and along with it which action to execute.